mount/mount_windows.go

```diff
@@ -82,8 +117,32 @@ func (m *Mount) GetParentPaths() ([]string, error) {
 
 // Unmount the mount at the provided path
 func Unmount(mount string, flags int) error {
+	mount = filepath.Clean(mount)
+
+	// Helpfully, both reparse points and symlinks look like links to Go
+	// Less-helpfully, ReadLink cannot return \\?\Volume{GUID} for a volume mount,
+	// and ends up returning the directory we gave it for some reason.
+	if mountTarget, err := os.Readlink(mount); err != nil {
+		return errors.Wrapf(err, "failed to readlink %s", mount)
+	} else if mount != filepath.Clean(mountTarget) {
+		// Directory symlink
+		return errors.Wrapf(bindUnmount(mount), "failed to bind-unmount from %s", mount)
+	}
```

Comment:
This could probably be made safer using `winio.DecodeReparsePoint`, so we don't treat a symlink to itself as a mount point. See https://github.com/microsoft/go-winio/blob/3fe6c526287328ec859e2c1258458b685198f204/backuptar/tar.go#L160-L171 for an example usage.